builtin/providers/aws/resource_aws_alb_test.go

```diff
@@ -45,6 +46,48 @@ func TestAccAWSALB_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSALB_generatedName(t *testing.T) {
+	var conf elbv2.LoadBalancer
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_alb.alb_test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSALBDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSALBConfig_generatedName(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBExists("aws_alb.alb_test", &conf),
+					resource.TestCheckResourceAttrSet("aws_alb.alb_test", "name"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSALB_namePrefix(t *testing.T) {
+	var conf elbv2.LoadBalancer
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_alb.alb_test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSALBDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSALBConfig_namePrefix(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSALBExists("aws_alb.alb_test", &conf),
+					resource.TestCheckResourceAttrSet("aws_alb.alb_test", "name"),
+					resource.TestMatchResourceAttr("aws_alb.alb_test", "name",
+						regexp.MustCompile("^tf-lb+")),
```

Comment:
It is basically working, but the `+` quantifier at the end doesn't make much sense as it belongs to `b`.
I assume `"^tf-lb.+"` is what you intended to type here? Perhaps `"^tf-lb"` would be sufficient and less greedy.